channel.py

```diff
@@ -464,17 +485,34 @@ def widen(self, flag=True):
     def narrow(self):
         self.widen(False)
 
+    def use_prefader_metering(self, flag=True):
+        self.meter_prefader = flag
+        self.prefader_button.set_active(flag)
+
     def read_meter(self):
         if not self.channel:
             return
         if self.stereo:
-            peak_left, peak_right, rms_left, rms_right = self.channel.kmeter
+            if self.meter_prefader:
+                peak_left, peak_right, rms_left, rms_right = self.channel.kmeter_prefader
+                self.channel.kmeter_postfader
+            else:
+                peak_left, peak_right, rms_left, rms_right = self.channel.kmeter_postfader
+                self.channel.kmeter_prefader
             self.meter.set_values(peak_left, peak_right, rms_left, rms_right)
         else:
-            peak, rms = self.channel.kmeter
+            if self.meter_prefader:
+                peak, rms = self.channel.kmeter_prefader
+                self.channel.kmeter_postfader
+            else:
+                peak, rms = self.channel.kmeter_postfader
+                self.channel.kmeter_prefader
             self.meter.set_values(peak, rms)
 
-        self.abspeak.set_peak(self.channel.abspeak)
+        if self.meter_prefader:
+            self.abspeak.set_peak(self.channel.abspeak_prefader)
+        else:
+            self.abspeak.set_peak(self.channel.abspeak_postfader)
```

Comment:
This can be shortened to:

```py
        self.meter.set_values(*(self.channel.kmeter_prefader if self.meter_prefader else
                              self.channel.kmeter_postfader))
        self.abspeak.set_peak(self.channel.abspeak_prefader if self.meter_prefader else
                              self.channel.abspeak_postfader)
```

:)